src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -848,6 +848,18 @@ leader_balancer::build_preference_index() {
         const auto& preference
           = topic.second.get_configuration().properties.leaders_preference;
         if (preference.has_value()) {
+            if (
+              should_sanction
+              && preference.value() != config::leaders_preference{}) {
+                vlog(
+                  clusterlog.warn,
+                  "A Redpanda Enterprise Edition license is required to use "
+                  "topic property \"redpanda.leaders.preference\" in topic {}. "
+                  "This property is being ignored.",
```

Comment:
I wonder if we should reuse the string?
```suggestion
                  R"(A Redpanda Enterprise Edition license is required to use "
                  "topic property "{}" in topic {}. This property is being ignored.)",
                  kafka::topic_property_leaders_preference,
```

Reply:
....except that ends up with a cyclic dependency with bazel.

Reply:
while i agree that using the variable name is better, are we good to include a kafka header? It feels off.

Reply:
> are we good to include a kafka header? It feels off.

Well, not as it currently is, as bazel will complain.

In general, cluster module does too much stuff, but splitting the dependencies here isn't worth it.